### PR TITLE
Added a feature to exit the app when permissions are denied

### PR DESCRIPTION
**Pull request by shailypreetham**
|PR checklist|Status| 
|-----|:------:| 
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/GLEXP-Team-onebillion-Hindi/issues/167)| 
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes| 
|Is code analysis task executed and issues resolved in the CPP/native code?|NA| 
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA| 
|Is code commenting done properly for the changes made?|Yes| 
|Is correct variable naming convention followed?|Yes| 
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes| 
|Are the message strings used in the application taken from strings.xml file?|NA| 
|Are the permissions which are not required removed from AndroidManifest.java file?|NA| 
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA| 
|Are the generated files correctly configured in .gitignore?|NA| 
|Are the release signing configurations removed from the gradle files?|NA| 
|Is commented code removed from all files?|NA| 
|Are updated images compressed (< 100 KB)?|NA| 
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java [hunk 1]
```diff
@@ -106,7 +106,7 @@
 
     private long backPressedTime;                                                                   // to record the time for back button.
 
-    AudioManager audioManager;                                                                      //declaring audio manager object.
+    AudioManager audioManagerSetHalf;                                                                      //declaring audio manager object.
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java]: Declare as volumeManager instead

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java [hunk 2]
```diff
@@ -158,6 +158,19 @@ public static boolean isSDKCompatible() {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
 
+        systemsManager = new OBSystemsManager(this);
+
+        if (!isTaskRoot()) {                                                                                        //if the activity is not task root, close it
+            onDestroy();
+            finish();
+            return;
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java:166]: return is redundant. Remove it.

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java [hunk 3]
```diff
@@ -158,6 +158,19 @@ public static boolean isSDKCompatible() {
...
+        }
+
+        if (getIntent().getBooleanExtra("Exit me", false)) {                                         //finishes the main activity if the permissions are denied
+            finish();
+            return;
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java]: return is redundant. Remove it

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java [hunk 4]
```diff
@@ -540,11 +553,15 @@ public void onBatteryStatusReceived(float level, boolean charging) {
     }
 
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        if (requestCode == REQUEST_EXTERNAL_STORAGE && grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+        if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED && grantResults[3] == PackageManager.PERMISSION_GRANTED && grantResults[4] == PackageManager.PERMISSION_GRANTED) {
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java]: Shouldn't it be as shown below?
grantResults[0] == PackageManager.PERMISSION_GRANTED && grantResults[1] == PackageManager.PERMISSION_GRANTED && grantResults[2] == PackageManager.PERMISSION_GRANTED

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

**Review by akashdabhi03 — DISMISSED** [review]

**Review by akashdabhi03 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**